subsys/bluetooth/host/att.c

```diff
@@ -2420,6 +2537,11 @@ static void att_chan_detach(struct bt_att_chan *chan)
 		k_sem_give(&chan->tx_sem);
 	}
 
+	/* Release pending buffers */
```

Comment:
In the ATT timeout handler the channel that the ATT timeout happened for will be detached, what about if you had other  ATT channels active at the same time? How would they be released?

Reply:
Timeout is per bearer, other channels will not be affected, provided they don't timeout as well.

Reply:
But you would call att_reset and notify gatt that the connection is disconnect, not the channel. Since att_reset is called the att->conn is set to NULL and bt_att_send will now return ENOTCONN for everything as far as I can tell.

Reply:
Well read it again:

```
	/* Don't notify if there are still channels to be used */
	if (!sys_slist_is_empty(&att->chans)) {
		return;
	}
```

Reply:
Right. missed that.